gokitmiddlewares/stalemiddleware/middleware.go

```diff
@@ -0,0 +1,39 @@
+package stalemiddleware
+
+import (
+	"context"
+	"time"
+
+	"github.com/arquivei/foundationkit/app"
+	"github.com/go-kit/kit/endpoint"
+)
+
+func New(c Config) endpoint.Middleware {
+	ch := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ch:
+				if !app.IsHealthy() {
```

Comment:
Nit: app might be set as Unhealthy by other mechanisms, and then receiving requests will override that mechanism and set it back as healthy again. A true solution would be a Health manager which could evaluate a series of rules (by means of delegates), but for this changeset I'd simply add a flag in this middleware indicating that it was it that set the app as Unhealthy.

Potential problem, though: it and another mechanism sets Unhealthy.

Reply:
The healthy flag is global and any part of the code could change it. Like a database conector middleware could intendify a bad connection and mark the do as unhealky.

I like of the idea of a manager. I have some ideas on how we could do it simply.

Reply:
I agree with Unhealthy being an unrecoverable state